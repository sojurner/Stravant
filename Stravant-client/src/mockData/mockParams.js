export const tokenParams = [
  'https://www.strava.com/oauth/token',
  {
    body:
      '{"client_id":28236,"client_secret":"829f12d9bdb8ce424edf87deff4551c117d3ed36","code":"r5423rffda4c234ctwtwert2ct"}',
    headers: { 'Content-Type': 'application/json' },
    method: 'POST'
  }
];

export const aggregateParams = [
  'https://www.strava.com/api/v3/athletes/28236/stats',
  {
    headers: {
      Authorization: 'Bearer a7eeea216a46a760a74c60acbffc3b55c66537c5',
      'Content-Type': 'application/json'
    },
    method: 'GET'
  }
];

export const weeklyParams = [
  'https://www.strava.com/api/v3/athlete/activities?before=1536123319&after=1536036919',
  {
    headers: {
      Authorization: 'Bearer a7eeea216a46a760a74c60acbffc3b55c66537c5',
      'Content-Type': 'application/json'
    },
    method: 'GET'
  }
];
