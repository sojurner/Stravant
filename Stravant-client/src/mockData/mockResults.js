export const aggregateResult = { Biked: 0, Ran: 6.472964574269732, Swam: 0 };

export const tokenResult = {
  access_token: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
  athlete: {
    badge_type_id: 0,
    city: null,
    country: null,
    created_at: '2018-09-04T22:53:48Z',
    email: 'pykim0591@gmail.com',
    firstname: 'Paul',
    follower: null,
    friend: null,
    id: 34515396,
    lastname: 'K',
    premium: false,
    profile:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg',
    profile_medium:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg',
    resource_state: 2,
    sex: 'M',
    state: null,
    summit: false,
    updated_at: '2018-09-07T12:20:05Z',
    username: null
  },
  token_type: 'Bearer'
};

export const weeklyResult = {
  Friday: 3784.3,
  Monday: 0,
  Saturday: 0,
  Sunday: 0,
  Thursday: 0,
  Tuesday: 0,
  Wednesday: 0
};
