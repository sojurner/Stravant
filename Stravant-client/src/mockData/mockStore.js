export const currentUser = {
  info: {
    accessToken: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
    userId: 34515396,
    firstName: 'Paul',
    gender: 'M',
    profilePic:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg'
  },
  totalStats: {},
  weeklyStats: {}
};

export const pomStatus = {
  pomStatus: true,
  lastPom: '',
  pomHistory: {
    'Tue, Sep 11, 2018 8:39 PM': { second: 19, minute: 0, hour: 0 },
    'Tue, Sep 11, 2018 8:40 PM': { second: 3, minute: 0, hour: 0 },
    'Tue, Sep 11, 2018 10:22 PM': { second: 3, minute: 0, hour: 0 }
  }
};

export const clubInfo = {
  clubInfo: { clubName: 'Stravant Club', clubId: 473825 },
  clubActivity: {
    mostRecent: {
      name: 'Pamela',
      distance: 4.270478558110628,
      activity: 'Morning Run'
    },
    Pamela: {
      totalDistance: 62.88999999999999,
      totalTime: 635,
      membersRecent: 'Afternoon Run'
    },
    Paul: { totalDistance: 6.46, totalTime: 78, membersRecent: 'Morning Run' },
    Jesse: {
      totalDistance: 277.78000000000003,
      totalTime: 5292,
      membersRecent: 'Morning Run'
    }
  }
};

export const mockStore = {
  ...currentUser,
  ...pomStatus,
  ...clubInfo
};
