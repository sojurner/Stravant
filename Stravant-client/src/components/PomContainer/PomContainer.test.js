import React from 'react';
import { mount } from 'enzyme';
import {
  PomContainer,
  mapDispatchToProps,
  mapStateToProps
} from './PomContainer';
import * as store from '../../mockData/mockStore';

describe('PomContainer', () => {
  let wrapper;
  let mockPomHistory;
  let mockRemove;

  beforeEach(() => {
    mockPomHistory = [
      {
        id: 473825,
        resource_state: 2,
        name: 'Stravant Club',
        profile_medium:
          'https://dgalywyr863hv.cloudfront.net/pictures/clubs/473825/10246543/1/medium.jpg',
        profile:
          'https://dgalywyr863hv.cloudfront.net/pictures/clubs/473825/10246543/1/large.jpg',
        cover_photo:
          'https://dgalywyr863hv.cloudfront.net/pictures/clubs/473825/10246574/1/large.jpg',
        cover_photo_small:
          'https://dgalywyr863hv.cloudfront.net/pictures/clubs/473825/10246574/1/small.jpg',
        sport_type: 'running',
        city: 'Denver',
        state: 'Colorado',
        country: 'United States',
        private: false,
        member_count: 4,
        featured: false,
        verified: false,
        url: 'stravant'
      }
    ];

    mockRemove = jest.fn();
  });

  it('should matchsnapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call removePom on click', () => {
    wrapper = mount(
      <PomContainer pomHistory={mockPomHistory} removePom={mockRemove} />
    );
    wrapper.find('i').simulate('click');
    expect(mockRemove).toHaveBeenCalled();
  });

  it('should map to the store properly', () => {
    const mockStore = store.pomStatus;

    const mapped = mapStateToProps(mockStore);

    expect(mapped.pomHistory).toEqual(mockStore.pomInfo.pomHistory);
  });

  it('should call dispatch function when using a function from mapDispatchtoProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.removePom();

    expect(mockDispatch).toHaveBeenCalled();
  });
});
