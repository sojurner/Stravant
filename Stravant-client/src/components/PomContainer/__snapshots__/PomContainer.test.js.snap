// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PomContainer should matchsnapshot 1`] = `undefined`;
